uint8[] id
int16[] position
